models/pwa.js

```diff
@@ -101,6 +101,12 @@ class Pwa {
     return this._manifest;
   }
 
+  generateEncodedStartUrl() {
+    const parsedUrl = URL.parse(this.absoluteStartUrl);
+    this.encodedStartUrl = encodeURIComponent(parsedUrl.hostname + parsedUrl.pathname);
+    return this.encodedStartUrl;
```

Comment:
We need to store as a DB indexed field it to be able to query PWAs by startURL, the idea is to use it soon to make PWA directory URLs human readable.